media/mod_sampledata/js/sampledata-process.js

```diff
@@ -3,78 +3,99 @@
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
-!(function ($) {
-	"use strict";
+Joomla = window.Joomla || {};
 
-	var inProgress = false;
-
-	var sampledataAjax = function(type, steps, step) {
-		if (step > steps) {
-			$('.sampledata-' + type + ' .row-title').append('<span class="icon-publish"> </span>');
-			inProgress = false;
-			return;
-		}
-		var stepClass = 'sampledata-steps-' + type + '-' + step,
-			$stepLi = $('<li class="' + stepClass + '"><p class="loader-image text-center"><img src="' + window.modSampledataIconProgress + '" width="30" height="30" ></p></li>'),
-			$progress = $(".sampledata-progress-" + type + " progress");
-
-		$("div.sampledata-progress-" + type + " ul").append($stepLi);
-
-		var request = $.ajax({
-			url: window.modSampledataUrl,
-			type: 'POST',
-			dataType: 'json',
-			data: {
-				type: type,
-				plugin: 'SampledataApplyStep' + step,
-				step: step
-			}
-		});
-		request.done(function(response){
-			$stepLi.children('.loader-image').remove();
-
-			if (response.success && response.data && response.data.length > 0) {
-				var success, value, resultClass, $msg;
-
-				// Display all messages that we got
-				for(var i = 0, l = response.data.length; i < l; i++) {
-					value   = response.data[i];
-					success = value.success;
-					resultClass = success ? 'success' : 'error';
-					$stepLi.append($('<div>', {
-						html: value.message,
-						'class': 'alert alert-' + resultClass,
-					}));
-				}
+!(function (Joomla, document) {
+	'use strict';
 
-				// Update progress
-				$progress.val(step/steps);
+	var inProgress = false;
 
-				// Move on next step
-				if (success) {
-					step++;
-					sampledataAjax(type, steps, step);
+	Joomla.sampledataAjax = function(type, steps, step) {
```

Comment:
you can do `const = sampledataAjax` here since we are inside an IIFE and that function is used only in this context

Reply:
This isn't ES6 mate